src/data/mv_polynomial/basic.lean

```diff
@@ -128,19 +128,25 @@ instance [comm_semiring S₁] [smul_zero_class R S₁] [smul_zero_class Rᵐᵒ
 add_monoid_algebra.is_central_scalar
 
 instance [comm_semiring R] [comm_semiring S₁] [algebra R S₁] : algebra R (mv_polynomial σ S₁) :=
+
 add_monoid_algebra.algebra
 
 -- Register with high priority to avoid timeout in `data.mv_polynomial.pderiv`
 instance is_scalar_tower' [comm_semiring R] [comm_semiring S₁] [algebra R S₁] :
   is_scalar_tower R (mv_polynomial σ S₁) (mv_polynomial σ S₁) :=
 is_scalar_tower.right
 
+instance is_scalar_tower_right [comm_semiring R] [comm_semiring S₁] [distrib_smul R S₁]
```

Comment:
This is a stronger assumption on `R` than needed. I'll split this to its own PR, and add the missing `smul_comm_class` too

Reply:
#19187